frontend/src/components/oidcauth/OauthPopup.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {PureComponent, ReactChild} from 'react';
+
+type Props = {
+  width: number;
+  height: number;
+  url: string;
+  title: string;
+  onClose: () => any;
+  onCode: (params: any) => any;
+  children?: ReactChild;
+}
+
+export default class OauthPopup extends PureComponent<Props> {
+
+  static defaultProps = {
+    onClose: () => {},
+    width: 500,
+    height: 500,
+    url: '',
+    title: ''
+  };
+
+  externalWindow: any;
+  codeCheck: any;
+
+  componentWillUnmount() {
+    if (this.externalWindow) {
+      this.externalWindow.close();
+    }
+  }
+
+  createPopup = () => {
+    const {url, title, width, height, onCode} = this.props;
+    const left = window.screenX + (window.outerWidth - width) / 2;
+    const top = window.screenY + (window.outerHeight - height) / 2.5;
+
+    const windowFeatures = `toolbar=0,scrollbars=1,status=1,resizable=0,location=1,menuBar=0,width=${width},height=${height},top=${top},left=${left}`;
+
+    this.externalWindow = window.open(
+        url,
+        title,
+        windowFeatures
+    );
+
+    const storageListener = () => {
+      console.log('storage listener trigerred');
+      try {
+        if (localStorage.getItem('auth_status')) {
+          console.log('trying to close window');
+          onCode(localStorage.getItem('auth_status'));
+          localStorage.removeItem('auth_status');
+          this.externalWindow.close();
+          window.removeEventListener('storage', storageListener);
+        }
+      } catch (e) {
+        console.log('handle error', e);
```

Comment:
You should lot this but maybe give it a bit more context, otherwise it seems like a debug log.